### PR TITLE
Call specFn and hookFn with received arguments

### PR DESCRIPTION
**Pull request by mucaho**
Some test frameworks (e.g. QUnit) supply additional arguments when
calling specFn.
Catch these arguments in the wrapper function and pass them onto
specFn and hookFn.

### CODE REVIEW

**Comment by christian-bromann** [comment]
> supply additional arguments when calling specFn

What kind of arguments are these? I don't think we should for instance propagate a `done` callback function since all commands run synchronous.

**Comment by mucaho** [comment]
> What kind of arguments are these?

For example `assert` in the following test code
```js
QUnit.test( "square()", function( assert ) {
  var result = Math.pow(2, 2);
  assert.equal( result, 4, "square(2) equals 4" );
});
```

> I don't think we should for instance propagate a done callback function since all commands run synchronous.

Would it hurt to include them? Would it still work if user accidentally called `done`?

**Comment by christian-bromann** [comment]
> Would it hurt to include them? Would it still work if user accidentally called done?

That's the thing I want to avoid being able to use that function since it is a bad pattern and still around for ages. There are promises now and synchronous execution doesn't require it at all. So being able to remove that is worth doing as it avoids confusion.

**Comment by mucaho** [comment]
Keep in mind it this could be an issue for other, future test frameworks too, but I guess this can be solved on a case-by-case basis.    
So should I change the implementation that only `assert` gets passed as an argument then?

**Comment by christian-bromann** [comment]
> this can be solved on a case-by-case basis

yup

> So should I change the implementation that only assert gets passed as an argument then?

That would be great

**Comment by mucaho** [comment]
Ok, now the methods filter out any not whitelisted arguments.
Addtionally whitelisted the `hooks` argument as needed in `QUnit.module`, e.g.
```js
QUnit.module("testModule", function(hooks) {
    hooks.beforeEach(function (assert) {
      assert.ok(true, "beforeEach called" );
    });

    QUnit.test("test", function (assert) {
      assert.equal(1, 1, "ok");
    });
});
```
Added tests.

File: index.js [hunk 1]
```diff
@@ -574,26 +578,32 @@ let runHook = function (hookFn, origFn, before, after, repeatTest = 0) {
  * @return {Function}             wrapped test function
  */
 let runSpec = function (specTitle, specFn, origFn, repeatTest = 0) {
+    // filter out arguments passed to specFn, don't allow callbacks
+    // as there is no need for user to call e.g. `done()`
+    const filterSpecArgs = (args) => {
```
Comment (christian-bromann) [inline index.js]: Let's define this function somewhere at the top of the file and reuse it everywhere in `runHook` and here

**Comment by mucaho** [comment]
Done

**Pull request merged**